### PR TITLE
Introduced GeneratorProperties as a thread-safe clone of System Properties

### PR DESCRIPTION
**Pull request by gndrm**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.4.x`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

see Issue #1597.

Implemented GeneratorProperties with a thread-local clone of SystemProperties to be used while Running generator-plugins.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/config/GeneratorProperties.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/config/GeneratorProperties.java]: For new files, there's no need to include line 3. Please refer to https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license/7301#7301 for more information.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/config/GeneratorProperties.java]: @gndrm after you remove line 3, we'll be happy to merge your contributions to the master.

Have a nice weekend.

Comment (gndrm) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/config/GeneratorProperties.java]: done, sorry for the late reply. was very short on time the last weeks.

**Comment by wing328** [comment]
@gndrm thanks for the PR, which has been merged into master.

**Comment by gndrm** [comment]
@wing328 i cannot reproduce your warning. i checked out the master today on a fresh system and ran
``
mvn clean install
mvn javadoc:javadoc
``

**Comment by wing328** [comment]
@gndrm 👌 

Thanks again for the contribution, which has been included in the 4.0.0-beta release: https://twitter.com/oas_generator/status/1079727020374806529.

Happy New Year and looking forward to more collaboration and contributions in 2019!

**Pull request merged**